### PR TITLE
feat: Don't hide errors with parsing `network config`.

### PR DESCRIPTION
**Pull request by pmnoxx**
`client config` contains a sections for `near-network` called `NearNetworkConfig`.

In function `validate()` of `NearNetworkConfig` we verify all argument. However, if any argument is invalid, we will print an error an continue running an normal. 

The proper solution is to, throw an error, and make let the caller know, that the `network_config` is invalid. Btw, we don't check whenever the number of peers allowed is to high. See https://github.com/near/nearcore/pull/5703

### CODE REVIEW

File: chain/network-primitives/src/types.rs [hunk 1]
```diff
@@ -584,15 +585,17 @@ impl NetworkConfig {
         }
     }
 
-    pub fn verify(&self) {
+    pub fn verify(&self) -> Result<()> {
```
Comment (mm-near) [inline chain/network-primitives/src/types.rs:588]: can you start a binary with mainnet/testnet config to see that these things don't crash ?

(or can you check the logs that none of these errors are printed?)

File: chain/network-primitives/src/types.rs [hunk 2]
```diff
@@ -603,20 +606,20 @@ impl NetworkConfig {
         }
 
         if self.safe_set_size <= self.minimum_outbound_peers {
-            error!(target: "network",
+            bail!(
```
Comment (mm-near) [inline chain/network-primitives/src/types.rs:609]: can you add a few tests to  test these conditiions where we bail now ?

**Review by mm-near — DISMISSED** [review]

File: chain/network-primitives/src/types.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/network-primitives/src/types.rs:588]: @mm-near Yes, I started testnet.

**Comment by pmnoxx** [comment]
```
@mm-near: chain/network-primitives/ nearcore/ neard/
@khorolets: chain/indexer/ tools/indexer/
```
@khorolets @mm-near Can you take a look?

**Review by khorolets — APPROVED** [review]

File: chain/network-primitives/src/types.rs (hunk 2 above)
Comment (mm-near) [reply chain/network-primitives/src/types.rs:609]: did you add the test?

**Review by mm-near — DISMISSED** [review]

File: chain/network-primitives/src/types.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network-primitives/src/types.rs]: I added a test last month. I rebased the PR.

**Pull request merged**